src/sdg/producten/models/product.py

```diff
@@ -270,9 +209,29 @@ def has_expired(self) -> bool:
             and (publication_date < reference_publication_date)
         )
 
+    @cached_property
+    def most_recent_version(self):
+        """
+        Returns the most recent `ProductVersie`.
+        """
+        # Check if prefetch cache is available from the manager.
+        result = getattr(self, "_most_recent_version", None)
+
+        # If not, retrieve it via the manager for consistancy.
+        if result is None:
+            p = self.__class__.objects.most_recent().filter(pk=self.pk).first()
+            result = p.most_recent_version
+
+        # If there's a version, return it.
+        if result:
+            return result[0]
+
+        return None
+
     @cached_property
     def laatste_versie(self):
         """:returns: Latest version (can be published, concept or scheduled) for this product."""
+        # TODO: Replace with `most_recent_version`.
```

Comment:
To make things constistant, this function should be replaced everywhere with `most_recent_version` but I didn't have time for that.